parsers/jscript.c

```diff
@@ -745,7 +745,10 @@ static void skipArgumentList (tokenInfo *const token, boolean include_newlines,
 		while (nest_level > 0 && ! isType (token, TOKEN_EOF))
 		{
 			readTokenFull (token, FALSE, repr);
-			if (isType (token, TOKEN_OPEN_PAREN))
+			if (isType (token, TOKEN_KEYWORD) && token->keyword == KEYWORD_function &&
+			    repr == NULL)
```

Comment:
BTW, there probably should never actually be anywhere we collect the argument list representation where a function definition is valid.